src/main.rs

```diff
@@ -258,6 +268,56 @@ fn main() {
             }
         }
 
+        "new" => {
+            let filetype = matches.value_of("FILETYPE").unwrap();
+            let mut filename = matches.value_of("FILENAME").unwrap();
+
+            match filetype {
+                "layout" => {
+                    if filename == "new_post.md" {
+                        filename = "new_layout.liquid";
+                    }
```

Comment:
Looks like clap has conditional defaults that we should take advantage of
https://docs.rs/clap/2.24.2/clap/struct.Arg.html#method.default_value_if

```rust
.arg(Arg::with_name("FILENAME")
    .help("Filter to create")
    .default_value_if("FILETYPE", Some("page"), "new_page.md"))
    .default_value_if("FILETYPE", Some("post"), "new_post.md"))
    .default_value_if("FILETYPE", Some("layout"), "new_layout.md"))
    .default_value("new.md"))
```